extensions/everhour/src/views/RecentTaskList.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from "react";
+import { List, Icon, showToast, ToastStyle } from "@raycast/api";
+import { TaskListItem } from "../components";
+import { getRecentTasks, getCurrentTimer } from "../api";
+import { Task } from "../types";
+import { createResolvedToast } from "../utils";
+
+export function RecentTaskList() {
+  const [activeTimerTaskId, setActiveTimerTaskId] = useState<null | string>(null);
+  const [tasks, setTasks] = useState<Array<Task>>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const refreshActiveTimer = async () => {
+    const toast = await showToast(ToastStyle.Animated, "Refreshing tasks");
+    try {
+      const activeTimer = await getCurrentTimer();
+      setActiveTimerTaskId(activeTimer);
+      createResolvedToast(toast, "Tasks refreshed").success();
+    } catch (error) {
+      createResolvedToast(toast, "Failed to refresh tasks").error();
+    }
+  };
+
+  const fetchTasks = async () => {
+    const tasksResp = await getRecentTasks();
+    setTasks(tasksResp);
+  };
+
+  useEffect(() => {
+    async function fetch() {
+      const toast = await showToast(ToastStyle.Animated, "Fetching tasks");
+      try {
+        await fetchTasks();
+        setIsLoading(false);
+        createResolvedToast(toast, "Tasks fetched").success();
+      } catch (error) {
+        const message = (error as { message: string }).message;
+        createResolvedToast(toast, message || "Failed to fetch projects").error();
+        setIsLoading(false);
+      }
+    }
+    fetch();
+  }, []);
+
+  useEffect(() => {
+    refreshActiveTimer();
+  }, [activeTimerTaskId]);
+
+  const renderTasks = () => {
+    if (tasks[0]) {
```

Comment:
this check isn't needed. You only call this function if `tasks[0]` is truthy in the check on line 73

Reply:
Don't agree, this function is called unconditionally in loc 80. Logic is copy pasted from https://github.com/ahoereth/raycast-extensions/blob/8e77b3a5a736d09d79e48f010d44d7df1e65da08/extensions/everhour/src/views/TaskList.tsx#L65